larray_editor/editor.py

```diff
@@ -275,6 +279,29 @@ def get_value(self):
         # already been destroyed, due to the Qt.WA_DeleteOnClose attribute
         return self.data
 
+    def add_widget_with_state_to_restore(self, widget_name, widget):
```

Comment:
Could you explain me why you need this two-step process (ie add widgets to a dict and then traverse that dict) instead of doing that in a "get_widgets_with_state_to_restore" method?

Reply:
How would write the "get_widgets_with_state_to_restore" method?